src/main.ts

```diff
@@ -277,18 +286,45 @@ class MealPluginSettingsTab extends PluginSettingTab {
 
         new Setting(containerEl)
             .setName('Shopping list ignore')
-            .setDesc('CSV list of ingredients to not add to the shopping list automatically')
-            .addText((text) => {
-                text.setPlaceholder('salt,pepper')
-                    .setValue(get(this.ctx.settings).shoppingListIgnore.join(','))
+            .setDesc('List of ingredients to not add to the shopping list automatically, one per line')
+            .addTextArea((textarea) => {
+                textarea.setPlaceholder('salt\npepper')
+                    .setValue(get(this.ctx.settings).shoppingListIgnore.join('\n'))
+                    .onChange(async (value) => {
+                        this.ctx.settings.update((s) => {
+                            s.shoppingListIgnore = value.split('\n').map(line => line.trim());
```

Comment:
Validate regex and maybe wildcard (potentially convert to regex and validate that) so errors don't go unnoticed